File: IS/front/src/ts/data/Country.ts
enum Country {
    UNITED_KINGDOM = 'UNITED_KINGDOM',
    SPAIN = 'SPAIN',
    THAILAND = 'THAILAND'
}

export const countryToString = new Map<Country, string>()
countryToString.set(Country.UNITED_KINGDOM, "Великобритания")
countryToString.set(Country.SPAIN, "Испания")
countryToString.set(Country.THAILAND, "Таиланд")

export default Country

File: back/src/main/java/com/example/back/model/dto/Country.java
package com.example.back.model.dto;

public enum Country {
    UNITED_KINGDOM,
    SPAIN,
    THAILAND;

    public static Country stringToCountry(String s) {
        return switch (s.toLowerCase()) {
            case "united_kingdom" -> UNITED_KINGDOM;
            case "spain" -> SPAIN;
            case "thailand" -> THAILAND;
            default -> null;
        };
    }
}